ssl/statem/statem_srvr.c

```diff
@@ -3591,6 +3621,21 @@ MSG_PROCESS_RETURN tls_process_client_certificate(SSL *s, PACKET *pkt)
     return ret;
 }
 
+#if !defined(OPENSSL_NO_COMP)
+MSG_PROCESS_RETURN tls_process_client_compressed_certificate(SSL *s, PACKET *pkt)
+{
+    MSG_PROCESS_RETURN ret = MSG_PROCESS_ERROR;
+    PACKET tmppkt;
+    BUF_MEM *buf = NULL;
+
+    if (tls13_process_compressed_certificate(s, pkt, &tmppkt, &buf) != MSG_PROCESS_ERROR)
+        ret = tls_process_client_certificate(s, &tmppkt);
+
+    BUF_MEM_free(buf);
```

Comment:
This is more of a niggle. Maybe it will be better to merge this function to prevent memory allocation inside other function and deallocation on this level. This code adds complexity to memory management.

Reply:
Possibly valid point, I would have to go look at the the `buf` handling.